### DISCUSSION TITLE
Does starship support "magic enter"?

### DISCUSSION THREAD

**Discussion by Kibouo** [starship/starship#2804] [Q&A]
Is it possible to configure starship such that modules are hidden by default, and only shown when the user presses Enter with empty input?

**Comment by chipbuster** [comment]
No, we don't support this feature at the moment.

Do you know of any frameworks that do support this? If so, would you mind linking them? I'm aware of a few that change the prompt in response to user input like pl10k, but I don't know of any that have this specific feature.

**Reply by Kibouo** [reply]
I used to use [geometry](https://github.com/geometry-zsh/geometry). It supports the `GEOMETRY_INFO` variable which defines commands to be ran after pressing enter with no input (tho it is lackluster as it strips newlines).

Currently I'm simply using starship with the [magic enter zsh plugin](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/magic-enter).

The use-case would be to have a minimalist prompt by default, while still having easy access to the info provided by modules.
